src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/configurations/BuckToGeneralTestEventsConverter.java

```diff
@@ -0,0 +1,188 @@
+import com.facebook.buck.test.result.type.ResultType;
+import com.google.common.collect.ImmutableSet;
+import com.intellij.execution.process.ProcessAdapter;
+import com.intellij.execution.process.ProcessEvent;
+import com.intellij.execution.process.ProcessHandler;
+import com.intellij.execution.process.ProcessOutputTypes;
+import com.intellij.execution.testframework.TestConsoleProperties;
+import com.intellij.execution.testframework.sm.runner.GeneralTestEventsProcessor;
+import com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter;
+import com.intellij.execution.testframework.sm.runner.events.TestFailedEvent;
+import com.intellij.execution.testframework.sm.runner.events.TestFinishedEvent;
+import com.intellij.execution.testframework.sm.runner.events.TestOutputEvent;
+import com.intellij.execution.testframework.sm.runner.events.TestStartedEvent;
+import com.intellij.execution.testframework.sm.runner.events.TestSuiteFinishedEvent;
+import com.intellij.execution.testframework.sm.runner.events.TestSuiteStartedEvent;
+import com.intellij.openapi.project.Project;
+import com.intellij.util.messages.MessageBusConnection;
+
+import org.jetbrains.annotations.NotNull;
+
+
+public class BuckToGeneralTestEventsConverter extends OutputToGeneralTestEventsConverter
+    implements TestRunStartedConsumer,
+    TestRunCompleteConsumer,
+    TestResultsAvailableConsumer,
+    BuckBuildStartConsumer,
+    BuckBuildEndConsumer,
+    CompilerErrorConsumer {
+
+  @NotNull
+  private final ProcessHandler myHandler;
+  @NotNull
+  private final Project mProject;
+  @NotNull
+  final MessageBusConnection mConnection;
+
+  public BuckToGeneralTestEventsConverter(
+      @NotNull String testFrameworkName,
+      @NotNull TestConsoleProperties consoleProperties,
+      @NotNull ProcessHandler handler,
+      @NotNull Project project) {
+    super(testFrameworkName, consoleProperties);
+    myHandler = handler;
+    mProject = project;
+    mConnection = mProject.getMessageBus()
+        .connect();
+    mConnection.subscribe(TestResultsAvailableConsumer.BUCK_TEST_RESULTS_AVAILABLE, this);
+    mConnection.subscribe(TestRunCompleteConsumer.BUCK_TEST_RUN_COMPLETE, this);
+    mConnection.subscribe(TestRunStartedConsumer.BUCK_TEST_RUN_STARTED, this);
+    mConnection.subscribe(BuckBuildStartConsumer.BUCK_BUILD_START, this);
+    mConnection.subscribe(BuckBuildEndConsumer.BUCK_BUILD_END, this);
+    mConnection.subscribe(CompilerErrorConsumer.COMPILER_ERROR_CONSUMER, this);
+    myHandler.addProcessListener(new ProcessAdapter() {
+      @Override
+      public void processTerminated(ProcessEvent event) {
+        mConnection.disconnect();
+      }
+    });
+  }
+
+  @Override
+  public void consumeTestRunStarted(long timestamp) {
+    final GeneralTestEventsProcessor processor = getProcessor();
+    if (processor == null) {
+      return;
+    }
+    processor.onUncapturedOutput(
+        "Test run started (" + timestamp + ").\n",
```

Comment:
timestamp doesn't look like valuable information here. Can you convert it to readable format?